relayer/src/link.rs

```diff
@@ -714,6 +722,57 @@ impl RelayPath {
         }
     }
 
+    /// Checks if a sent packet has been received on destination.
+    fn send_packet_received_on_dst(&self, packet: &Packet) -> Result<bool, LinkError> {
+        let unreceived_packet =
+            self.dst_chain()
+                .query_unreceived_packets(QueryUnreceivedPacketsRequest {
+                    port_id: self.dst_port_id().to_string(),
+                    channel_id: self.dst_channel_id().to_string(),
+                    packet_commitment_sequences: vec![packet.sequence.into()],
+                })?;
+
+        Ok(unreceived_packet.is_empty())
+    }
+
+    /// Checks if a packet commitment has been cleared on source.
+    /// The packet commitment is cleared when either an acknowledgment or a timeout is received on source.
+    fn send_packet_commitment_cleared_on_src(&self, packet: &Packet) -> Result<bool, LinkError> {
+        let (bytes, _) = self.src_chain().build_packet_proofs(
+            PacketMsgType::Recv,
+            self.src_port_id(),
+            self.src_channel_id(),
+            packet.sequence,
+            Height::zero(),
+        )?;
+
+        Ok(bytes.is_empty())
+    }
+
+    /// Checks if a send packet event has already been handled (e.g. by another relayer).
+    fn send_packet_event_handled(&self, sp: &SendPacket) -> Result<bool, LinkError> {
+        Ok(self.send_packet_received_on_dst(&sp.packet)?
+            || self.send_packet_commitment_cleared_on_src(&sp.packet)?)
+    }
+
+    /// Checks if an acknowledge packet has been received on source chain.
```

Comment:
Nit: This method can be confusing. Both the comment and method name talk about the source chain, but the query is against the destination chain.

Reply:
its the packet's source chain...which is the destination chain for the relay path that sends the ack. Definitely the comment should be clearer than it is.

Reply:
f915d886c2cb2ffcf006c8ac451b97055e09cb45